Improvements. Move web chat to React. Add sanic web server. Add black  formatter, Add pre-commit hooks. Add github workflow. Refactoring.

ivan-karlovich opened the pull request:
Improvements. Move web chat to React. Add sanic web server. Add black  formatter, Add pre-commit hooks. Add github workflow. Refactoring.

RomanPlusPlus commented on ai_replica/skills/get_crypto_data.py:
```diff
@@ -1,6 +1,10 @@
 """A skill that allows the replica to get the current cryptocurrencies data."""
+""" 
+    The skill is available as Rasa action. See action_price_of_thing. 
+    TODO: remove this skill? 
```
Good catch! No reason to duplicate them

RomanPlusPlus commented on README.md:
```diff
@@ -68,33 +93,49 @@ Then you can launch all the servers by running:
 
 `make start_all`
 
-The UI chat will be available at `http://localhost:8000`.
```
I've tried to chat with the bot immediately after the web UI opened, but got the following error:
`requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8002): Max retries exceeded with url: /webhooks/rest/webhook (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7629e0d9a0>: Failed to establish a new connection: [Errno 111] Connection refused')) `

But if I wait for some time, and then start chatting, there is no problem.

As I understand, the root cause of the problem is as follows:
the Web UI page opens before the Rasa server is fully up and running. Likely due to the loading Spacy stuff, which takes some time. 

The simplest solution would be to add to the readme that the user has to wait a few minutes until the Rasa server is fully up.

ivan-karlovich replied:
The note is added to README.md.

The following issue is created to keep in mind the possibility of displaying UI loader till the server is up: https://github.com/AI-replica/AI-replica/issues/13

RomanPlusPlus reviewed (changes requested):
Sorry for the delay with the review (was moving to another apartment). 

These intents seem to be not working (s. the attached image):

- action_translate_from_to
- action_price_of_thing

Aside from that, LGTM.

![Screenshot](https://user-images.githubusercontent.com/19929162/189828523-e55ef976-d20a-4b66-a293-27753e3f6316.png)

marcobaturan commented:
Can you consider putting a 'wait' function with advice to users like:
'Please, wait until services are completed and charged'
Then, the massage disappear and continued. But you will need a flow based
in detect the fully up and loading procedures, then start to interact.

El mar, 13 sept 2022 a las 8:43, Roman Sitelew ***@***.***>)
escribió:

> ***@***.**** requested changes on this pull request.
>
> Sorry for the delay with the review (was moving to another apartment).
>
> These intents seem to be not working (s. the attached image):
>
>    - action_translate_from_to
>    - action_price_of_thing
>
> Aside from that, LGTM.
>
> [image: Screenshot]
> <https://user-images.githubusercontent.com/19929162/189828523-e55ef976-d20a-4b66-a293-27753e3f6316.png>
> ------------------------------
>
> In ai_replica/skills/get_crypto_data.py
> <https://github.com/AI-replica/AI-replica/pull/12#discussion_r968505433>:
>
> > @@ -1,6 +1,10 @@
>  from ai_replica.utils.nlp import is_some_keyword_in_text
>
>  """A skill that allows the replica to get the current cryptocurrencies data."""
> +"""
> +    The skill is available as Rasa action. See action_price_of_thing.
> +    TODO: remove this skill?
>
> Good catch! No reason to duplicate them
> ------------------------------
>
> In README.md
> <https://github.com/AI-replica/AI-replica/pull/12#discussion_r968518193>:
>
> > @@ -68,33 +93,49 @@ Then you can launch all the servers by running:
>
>  `make start_all`
>
> -The UI chat will be available at `http://localhost:8000`.
>
> I've tried to chat with the bot immediately after the web UI opened, but
> got the following error:
> requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost',
> port=8002): Max retries exceeded with url: /webhooks/rest/webhook (Caused
> by NewConnectionError('<urllib3.connection.HTTPConnection object at
> 0x7f7629e0d9a0>: Failed to establish a new connection: [Errno 111]
> Connection refused'))
>
> But if I wait for some time, and then start chatting, there is no problem.
>
> As I understand, the root cause of the problem is as follows:
> the Web UI page opens before the Rasa server is fully up and running.
> Likely due to the loading Spacy stuff, which takes some time.
>
> The simplest solution would be to add to the readme that the user has to
> wait a few minutes until the Rasa server is fully up.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AI-replica/AI-replica/pull/12#pullrequestreview-1104266662>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVYCAID5PAYLFZ72GLKRU3V6APCNANCNFSM573PPRWA>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>


-- 

<https://about.me/marco.baturan?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=gmail_api&utm_content=thumb>
Marco Baturan
about.me/marco.baturan
<https://about.me/marco.baturan?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=gmail_api&utm_content=thumb>
*Un saludo/Kind regards.*

ivan-karlovich commented:
> Can you consider putting a 'wait' function with advice to users like: 'Please, wait until services are completed and charged' Then, the massage disappear and continued. But you will need a flow based in detect the fully up and loading procedures, then start to interact. El mar, 13 sept 2022 a las 8:43, Roman Sitelew ***@***.***>) escribió:
> […](#)
> ***@***.**** requested changes on this pull request. Sorry for the delay with the review (was moving to another apartment). These intents seem to be not working (s. the attached image): - action_translate_from_to - action_price_of_thing Aside from that, LGTM. [image: Screenshot] <https://user-images.githubusercontent.com/19929162/189828523-e55ef976-d20a-4b66-a293-27753e3f6316.png> ------------------------------ In ai_replica/skills/get_crypto_data.py <[#12 (comment)](https://github.com/AI-replica/AI-replica/pull/12#discussion_r968505433)>: > @@ -1,6 +1,10 @@ from ai_replica.utils.nlp import is_some_keyword_in_text """A skill that allows the replica to get the current cryptocurrencies data.""" +""" + The skill is available as Rasa action. See action_price_of_thing. + TODO: remove this skill? Good catch! No reason to duplicate them ------------------------------ In README.md <[#12 (comment)](https://github.com/AI-replica/AI-replica/pull/12#discussion_r968518193)>: > @@ -68,33 +93,49 @@ Then you can launch all the servers by running: `make start_all` -The UI chat will be available at `http://localhost:8000`. I've tried to chat with the bot immediately after the web UI opened, but got the following error: requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8002): Max retries exceeded with url: /webhooks/rest/webhook (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7629e0d9a0>: Failed to establish a new connection: [Errno 111] Connection refused')) But if I wait for some time, and then start chatting, there is no problem. As I understand, the root cause of the problem is as follows: the Web UI page opens before the Rasa server is fully up and running. Likely due to the loading Spacy stuff, which takes some time. The simplest solution would be to add to the readme that the user has to wait a few minutes until the Rasa server is fully up. — Reply to this email directly, view it on GitHub <[#12 (review)](https://github.com/AI-replica/AI-replica/pull/12#pullrequestreview-1104266662)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABVYCAID5PAYLFZ72GLKRU3V6APCNANCNFSM573PPRWA> . You are receiving this because your review was requested.Message ID: ***@***.***>
> -- <https://about.me/marco.baturan?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=gmail_api&utm_content=thumb> Marco Baturan about.me/marco.baturan <https://about.me/marco.baturan?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=gmail_api&utm_content=thumb> *Un saludo/Kind regards.*

The following issue is created to keep in mind the possibility of displaying UI loader till the server is up: https://github.com/AI-replica/AI-replica/issues/13

ivan-karlovich commented:
> These intents seem to be not working (s. the attached image):
> 
> * action_translate_from_to
> * action_price_of_thing

I have fixed the action "action_price_of_thing" and added more samples. Though the intents should work in more cases but need to keep in mind that they are still not generalized very well. I guess more examples are required. But for some cases they work.
![image](https://user-images.githubusercontent.com/19314095/190872988-990e4292-879c-467f-a376-c8e49c21d314.png)

RomanPlusPlus reviewed (approved):
LGTM.

Regarding the search idea:

I have some experience with implementing a very similar functionality with SpaCy and a few other tools, could apply it here too. But the best it can do is to return the most relevant parts of the unstructured text, which will not be good enough in the chatbot context. Maybe we can leapfrog into more advanced stuff, as you suggested. 

![Screenshot](https://user-images.githubusercontent.com/19929162/192043270-ce16fde4-f2cf-4411-bc65-36732f50017b.png)

ivan-karlovich commented:
@RomanPlusPlus, can this PR be merged?

ivan-karlovich commented:
> "Maybe we can leapfrog into more advanced stuff, as you suggested."

I guess we can try sentence embeddings. In particular, I am planning to check sbert model.
